panoptes_aggregation/extractors/poly_line_text_extractor.py

```diff
@@ -95,15 +95,19 @@ def poly_line_text_extractor(classification, dot_freq='word', gold_standard=Fals
                         dy = y_fit[-1] - y_fit[0]
                         slope = np.rad2deg(np.arctan2(dy, dx))
                     except np.RankWarning:
-                        # rotate by 90 before fitting
-                        x_tmp = -np.array(y)
-                        y_tmp = np.array(x)
-                        fit = np.polyfit(x_tmp, y_tmp, 1)
-                        y_fit = np.polyval(fit, [x_tmp[0], x_tmp[-1]])
-                        dx = x_tmp[-1] - x_tmp[0]
-                        dy = y_fit[-1] - y_fit[0]
-                        # rotate by -90 to bring back into correct coordiates
-                        slope = np.rad2deg(np.arctan2(dy, dx)) - 90
+                        try:
+                            # rotate by 90 before fitting
+                            x_tmp = -np.array(y)
+                            y_tmp = np.array(x)
+                            fit = np.polyfit(x_tmp, y_tmp, 1)
+                            y_fit = np.polyval(fit, [x_tmp[0], x_tmp[-1]])
+                            dx = x_tmp[-1] - x_tmp[0]
+                            dy = y_fit[-1] - y_fit[0]
+                            # rotate by -90 to bring back into correct coordiates
+                            slope = np.rad2deg(np.arctan2(dy, dx)) - 90
+                        except np.RankWarning:
```

Comment:
Not blocking - Will this swallow other error conditions that we might to know about?

Also it's normally computationally cheaper to test for the error condition (if you know it) rather than raise an exception

Reply:
This particular case was a line of exactly length 0, but I expect if the length was very small it could also lead to this condition, in either case I want this value to default to 0.

The reducer has a configuration option for filtering out short lines (typically lines sorter than 10px are just noise anyways).